.github/workflows/Regression.yml

```diff
@@ -65,11 +65,32 @@ jobs:
           key: ${{ github.job }}
           save: ${{ github.ref == 'refs/heads/main' || github.repository != 'duckdb/duckdb' }}
 
-      - name: Build
+      - name: Checkout Private Regression
+        if: ${{ github.repository == 'duckdb/duckdb' && github.ref == 'refs/heads/main' }}
+        uses: actions/checkout@v4
+        with:
+          repository: duckdblabs/fivetran_regression
+          ref: main
+          token: ${{ secrets.DUCKDBLABS_BOT_TOKEN }}
+          path: benchmark/fivetran
+
+      - name: Build Current and Base Branch
+        if: ${{ !(github.repository == 'duckdb/duckdb' && github.ref == 'refs/heads/main') }}
         shell: bash
         run: |
-          make
           git clone --branch ${{ env.BASE_BRANCH }} https://github.com/duckdb/duckdb.git --depth=1
+          make
+          cd duckdb
+          make
+          cd ..
+
+      - name: Build Main and Previous Successful Regression Hash
+        if: ${{ github.repository == 'duckdb/duckdb' && github.ref == 'refs/heads/main' }}
+        shell: bash
+        run: |
+          git clone https://github.com/duckdb/duckdb.git
+          git checkout $(gh run list --repo duckdb/duckdb --branch=main --workflow=Regression --status=success --json=headSha --limit=1 --jq '.[0].headSha')
```

Comment:
This is cute, this is avoiding keeping keeping explicit state by asking github actions for the state.

I though about adding also `--event repository_dispatch`, that given stuff is dispatched only every night, it means "last successful nightly". Unsure, probably fine as is. Maybe a comment would be nice.